### DISCUSSION TITLE
Is there any standardised way of indexing atoms

### DISCUSSION THREAD

**Discussion by syedzayyan** [rdkit/rdkit#6714] [FAQ] [answered]
Hello All!

I was following the RDKit Cookbook and tried to draw Caffeine using the given code:

<img width="271" alt="Screenshot 2023-09-12 at 12 01 13" src="https://github.com/rdkit/rdkit/assets/44922710/cc67d9f3-5835-4b4b-bb29-7e40ce0c9629">

This is the result from that. The code I use is this:

```
def mol_with_atom_index(mol):
    for atom in mol.GetAtoms():
        atom.SetAtomMapNum(atom.GetIdx())
    return mol
```

Then I was reading this paper:

<img width="317" alt="Screenshot 2023-09-12 at 12 02 05" src="https://github.com/rdkit/rdkit/assets/44922710/d0a86a3e-f98c-4721-ba54-701b80c9bf77">

Now they index their atoms differently here in this paper: https://www.cell.com/structure/pdfExtended/S0969-2126(17)30192-2

Now, my question is there any standard practise when indexing atoms and what could be the possible code for indexing atoms the 'right' way.

**Comment by bp-kelley** [comment, marked as answer]
To my knowledge, unless you are (1) explicitly talking about using IUPAC rules for indexing and naming or (2) using something like the the RCSB/PDB monomer library for things like creating structure submissions, there is no conventional way to index atoms.

REFS:
https://www.cas.org/sites/default/files/documents/indexguideapp.pdf
https://www.rcsb.org/ligand/LIB

**Reply by bp-kelley** [reply]
I will also point out that SetAtomMap(0) removes the atom map, you probably want to have one indexed atoms.

**Reply by syedzayyan** [reply]
I was somewhat asking IUPAC indexing. I am assuming the paper was using IUPAC as well since it matches the pubchem IUPAC name. Thank you very much for the resources.

**Reply by bp-kelley** [reply]
You can get the IUPAC locants using something like opsin if you have the IUPAC names.

https://opsin.ch.cam.ac.uk

But these don't seem to line up with the indexes used above.
